lib/driverProviders/attachSession.js

```diff
@@ -0,0 +1,58 @@
+/*
+ *  This is an implementation of the Attach Session Driver Provider.
+ *  It is responsible for setting up the account object, tearing
+ *  it down, and setting up the driver correctly.
+ */
+
+
+var util = require('util'),
+    q = require('q'),
+    DriverProvider = require('./driverProvider'),
+    log = require('../logger'),
+    webdriver = require('selenium-webdriver'),
+    executors = require('selenium-webdriver/executors');
+
+
+var AttachedSessionDriverProvider = function(config) {
+  DriverProvider.call(this, config);
+};
+util.inherits(AttachedSessionDriverProvider, DriverProvider);
+
+/**
+ * Configure and launch (if applicable) the object's environment.
+ * @public
+ * @return {q.promise} A promise which will resolve when the environment is
+ *     ready to test.
+ */
+AttachedSessionDriverProvider.prototype.setupEnv = function() {
+  log.puts('Using the selenium server at ' + this.config_.seleniumAddress);
+  log.puts('Using session id - ' + this.config_.seleniumSessionId);
+  return q.fcall(function() {});
```

Comment:
I could change that to `q(undefined)`. I was basing the `setupEnv` off of hosted.js.